src/linux/monitor.rs

```diff
@@ -118,16 +118,16 @@ where
         }
     }
 
-    fn remove_device(&mut self, path: OsString) {
-        if let Some(runloop) = self.runloops.remove(&path) {
+    fn remove_device(&mut self, path: &OsString) {
+        if let Some(runloop) = self.runloops.remove(path) {
             runloop.cancel();
         }
     }
 
     fn remove_all_devices(&mut self) {
         while !self.runloops.is_empty() {
             let path = self.runloops.keys().next().unwrap().clone();
```

Comment:
I imagine we don't need `clone()` here now? (I haven't tried it, but this should already be a reference, so we should just be able to pass it to `remove_device` directly)

Reply:
No, since `remove_device` borrows `self` mutably, we end up with a multi-level immutable borrow out of a mutable object, which is verboten. (`rustc --explain E0502`) . Since there's no mutable way to get keys out of a HashMap (`self.runloops`), we're stuck cloning so that Rust can guarantee we won't change the keys out from underneath the iteration.